packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -279,6 +284,8 @@ class _EditableTextWidget extends LeafRenderObjectWidget {
     }
 
     String text = value.text;
+    if (hideText)
+      text = new String.fromCharCodes(new List<int>.filled(text.length, 0x2022));
```

Comment:
The normal thing for password fields is to show one bullet per character.  That way typing and deleting have good visual feedback.